nix/rust.nix

```diff
@@ -96,76 +107,81 @@ in {
     '';
   };
 
-  plonk_wasm = let
+  plonk_wasm =
+    let
 
-    lock = ../src/lib/crypto/kimchi_bindings/wasm/Cargo.lock;
+      lock = ../src/lib/crypto/kimchi_bindings/wasm/Cargo.lock;
 
-    deps = builtins.listToAttrs (map (pkg: {
-      inherit (pkg) name;
-      value = pkg;
-    }) (builtins.fromTOML (builtins.readFile lock)).package);
+      deps = builtins.listToAttrs (map
+        (pkg: {
+          inherit (pkg) name;
+          value = pkg;
+        })
+        (builtins.fromTOML (builtins.readFile lock)).package);
 
-    rustPlatform = rustPlatformFor final.kimchi-rust-wasm;
+      rustPlatform = rustPlatformFor final.kimchi-rust-wasm;
 
-    wasm-bindgen-cli = rustPlatform.buildRustPackage rec {
+      wasm-bindgen-cli = rustPlatform.buildRustPackage rec {
 
-      pname = "wasm-bindgen-cli";
-      version = deps.wasm-bindgen.version;
-      src = final.fetchCrate {
-        inherit pname version;
-        sha256 = "sha256-BQ8v3rCLUvyCCdxo5U+NHh30l9Jwvk9Sz8YQv6fa0SU=";
-      };
+        pname = "wasm-bindgen-cli";
+        version = deps.wasm-bindgen.version;
+        src = final.fetchCrate {
+          inherit pname version;
+          sha256 = "sha256-BQ8v3rCLUvyCCdxo5U+NHh30l9Jwvk9Sz8YQv6fa0SU=";
+        };
 
-      cargoSha256 = "sha256-mP85+qi2KA0GieaBzbrQOBqYxBZNRJipvd2brCRGyOM=";
-      nativeBuildInputs = [ final.pkg-config ];
+        cargoSha256 = "sha256-mP85+qi2KA0GieaBzbrQOBqYxBZNRJipvd2brCRGyOM=";
+        nativeBuildInputs = [ final.pkg-config ];
 
-      buildInputs = with final;
-        [ openssl ] ++ lib.optionals stdenv.isDarwin [
-          curl
-          darwin.apple_sdk.frameworks.Security
-        ];
+        buildInputs = with final;
+          [ openssl ] ++ lib.optionals stdenv.isDarwin [
+            curl
+            darwin.apple_sdk.frameworks.Security
+            libiconv
```

Comment:
This change adds `libiconv` to inputs of `wasm-bindgen-cli`, which compiles fine as-is. The proper patch would be

```
diff --git a/nix/rust.nix b/nix/rust.nix
index 4a3b67dd30..4060cdcb33 100644
--- a/nix/rust.nix
+++ b/nix/rust.nix
@@ -72,6 +72,7 @@ in {
     ];
     sourceRoot = "source/lib/crypto/kimchi_bindings/stubs";
     nativeBuildInputs = [ final.ocamlPackages_mina.ocaml ];
+    buildInputs = with final; lib.optional stdenv.isDarwin libiconv;
     cargoLock = let fixupLockFile = path: builtins.readFile path;
     in {
       lockFileContents =
@@ -139,6 +140,7 @@ in {
     ];
     sourceRoot = "source/lib/crypto/kimchi_bindings/wasm";
     nativeBuildInputs = [ final.wasm-pack wasm-bindgen-cli ];
+    buildInputs = with final; lib.optional stdenv.isDarwin libiconv;
     cargoLock.lockFile = lock;
     cargoLock.outputHashes = narHashesFromCargoLock lock;
```

Reply:
This patch doesn't fix the error for me -- only my patch does

Reply:
ah the union of these two patches is good 👍